recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -66,16 +66,20 @@ def requirements(self):
             self.requires("boost/1.80.0")
 
     def validate(self):
-        if self.info.settings.arch != "x86_64":
-            raise ConanInvalidConfiguration(f"{self.ref} doesn't support architecture : {self.info.settings.arch}")
-
         if self.info.settings.compiler.cppstd:
             check_min_cppstd(self, self._minimum_cpp_standard)
         check_min_vs(self, "192")
 
         if self.settings.os != "Linux" and self.options.shared:
             raise ConanInvalidConfiguration(f"{self.ref} supports building shared libraries only on Linux")
 
+        if not self.options["grpc"].cpp_plugin:
```

Comment:
```suggestion
        if not self.dependencies["grpc"].options.cpp_plugin:
```
self.dependencies is supported by both COnan v1 and v2, but self.options works only with Conan v1